migrator/src/main/java/com/redhat/lightblue/migrator/Breakpoint.java

```diff
@@ -86,8 +88,10 @@ public void checkpoint() {
         } else {
             if(waiting) {
                 synchronized(this) {
-                    ran=true;
-                    notify();
+                    if (waiting) {
```

Comment:
'waiting' isn't declared as 'volatile'. It should be volatile, it doesn't change whether notify is called or not.